include/xsparse/level_capabilities/co_iteration.hpp

```diff
@@ -30,6 +59,26 @@ namespace xsparse::level_capabilities
             }
         }
 
+        constexpr auto ordered_levels() const noexcept
```

Comment:
@hameerabbasi and @bharath2438 does this look fine in your opinion to get tuple of booleans corresponding to ordered levels?

Reply:
Yes, looks fine. Is there any way to do away with `decay_t` by passing in the right argument to `std::apply`?

Reply:
I don't think so? What would "the right argument" to `std::apply` be? We only have access to `m_levelsTuple` and `m_comparisonHelper` in this setting.

Reply:
IIRC, `decay_t` is used for stripping off the reference (or const) from the type, so I was wondering if it was really necessary, or would `std::make_tuple(decltype(levels)::LevelProperties::is_ordered...)` be sufficient.

Reply:
Ah I see, yeah I tried that before (and just now again), but the issue is that:

```
In file included from /Users/adam2392/Documents/xsparse/test/source/coiteration_test.cpp:17:
/Users/adam2392/Documents/xsparse/include/xsparse/level_capabilities/co_iteration.hpp:77:25: error: type 'decltype(levels)' (aka 'const xsparse::levels::dense<std::tuple<>, unsigned long, unsigned long> &') cannot be used prior to '::' because it has no members
                        decltype(levels)::LevelProperties::is_ordered...);
```